### PR TITLE
feat(stdlib/pagerduty): add custom_details field support

### PR DESCRIPTION
**Pull request by alespour**
Extends Flux to partially satisfy https://github.com/influxdata/feature-requests/issues/300.

Since it is not possible to extend existing `sendEvent` and `endpoint` with additional parameters (and/or behavior) and preserve API backward compatibility at the same, they are left unchanged and new pair of functions `sendEventEx` and `endpointEx` are introduced, doing pretty much the same but with additional fields `custom_details` and `component` (which was documented in the code before but not implemented).

There are two reasons why existing functions cannot be extended without breaking backward compatibility. First, record type has no default value (`{}` infers empty record type), second, `exists` operator does not work for records other than retrieved from InfluxDB , so it is unusable in `pagerduty.endpoint` therefore user `mapFn` function must built record with all fields passed to `sendEvent`.

### Done checklist
- [x] Test cases written

### CODE REVIEW

**Comment by ivankudibal** [comment]
merge depends on https://github.com/influxdata/flux/issues/3461

**Comment by nathanielc** [comment]
Hi @ivankudibal and @alespour I just added a feature to Flux that should allow you to update this PR in a backwards compatible way. See https://github.com/influxdata/flux/pull/4041 In short you can set the default value to `record.any` instead of `{}` and it will allow any record to be passed in and default to the empty record.

**Comment by alespour** [comment]
Hi @nathanielc, this is very useful. Function `sendEvent` arguments can be safely extended with record type args. 

There is still however compatibility issue with `endpoint` function. User calls it with `mapFn` function argument. This function returns a record whose fields are then used as arguments to `sendEvent` function. We've tried before to use `exists` operator to check if given field exists in the record and if not, provide some default, but it does not work (I think it was Chris Wolff who told me that this operator works only on records returned from InfluxDB).
So, existing user code where `mapFn` does not return the new fields would fail.

I've added `builtin get : (r: A, key: string, default: B) => B where A: Record` method to `record` pkg that could be used to solve this, I've tested and it works. ``. Draft PR #4049.

**Comment by alespour** [comment]
So, if the the new function `record.get` in PR #4049 is acceptable solution, I will complete that PR with tests and make it avail for review. And after it's merged, I could finish this PR.

**Comment by nathanielc** [comment]
> We've tried before to use exists operator to check if given field exists in the record and if not, provide some default, but it does not work 

This should work when using `record.any`. Can you share what you tried and how it failed?

**Comment by alespour** [comment]
If I add a line like
```
   customDetails: if exists obj.customDetails then obj.customDetails else record.any
```
after https://github.com/influxdata/flux/blob/5cd8f8c037857879511306c03e702da81d255741/stdlib/pagerduty/pagerduty.flux#L227

I get `record is missing label customDetails`. `obj` is a record returned by user supplied function, so this would break existing user code.

**Comment by nathanielc** [comment]
@alespour That looks like a good solution. We had a team discussion to see if there are other options, but we agree that `record.get` is the best short term solution.

However we want to add some exceptions to the `record.get` implementation.

* It needs to be experimental and with documentation that we plan to remove/update it once we fix `exists` to work with all records
* It needs to have documentation that states that the key must be a string literal, like you have shown. But uses that would pass in a dynamic string value should be forbidden, unfortunately it is not easy to enforce this in the code.

In short adding `record.get` is a hack  as it essentially introduces reflection into the Flux engine. Once we fix the `exists` behavior we expect that we will be able to update the implementation of `record.get` to not need this psuedo-reflection. 

Thanks for understanding as we work through this difficult challenges in designing a language.

File: stdlib/pagerduty/pagerduty.flux [hunk 1]
```diff
@@ -143,7 +149,16 @@ sendEvent = (
         "Accept": "application/vnd.pagerduty+json;version=2",
         "Content-Type": "application/json",
     }
-    enc = json.encode(v: data)
+    enc = if customDetails == record.any then
```
Comment (nathanielc) [inline stdlib/pagerduty/pagerduty.flux:152]: why do we need this conditional? Is there an error encoding an empty JSON object as the custom details?

**Review by nathanielc — COMMENTED** [review]
LGTM I have a one question
